root/static/scripts/edit/components/FormRowNameWithGuessCase.js

```diff
@@ -99,6 +99,8 @@ component FormRowNameWithGuessCase(
   label: React$Node = addColonText(l('Name')),
 ) {
   const inputRef = React.useRef<HTMLInputElement | null>(null);
+  const [nameBeforePreview, setNameBeforePreview] =
+    React.useState<string | null>(null);
```

Comment:
Using `React.useState` here is fine, though I'd suggest storing the preview in state rather than the name-before-preview. The reason for this is that we will eventually want to display edit previews on all entity pages (similar to the release editor), and don't want to generate those whenever the GC button is hovered over.

I believe we could accomplish this by just adding a `preview` prop to `FormRowText` which overrides the field value if specified.

Since I wasn't certain at first if this idea made sense, I tried it locally, so I'll provide the hastily-written patch I had (though you may want to review it for issues yourself :)). https://gist.github.com/mwiencek/2b36babb8aa2ec8ec0df7a1e7fb53f43

Reply:
Thanks, much cleaner, and I didn't notice any problems! I went ahead and renamed the state variable to just be `preview` for both components and updated the "copy sort name" button to also show a preview.